### ISSUE TITLE
A JavaScript Error Occurred in the Main Process

### ISSUE THREAD

**Issue by colingzesh** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#681]
**OS:**
Windows 10 64 Bit
**Issue Descriptions:**
I cannot open or reinstall the client.  Every time I do, I get an error message that says"A JavaScript Error Occurred in the Main Process - Uncaught Exception: Syntax Error: Unexpected Token
**Steps to Reproduce:**
It was working fine this morning until I installed Itunes, Audible Download Manager, and AudibleManager.  However, I doubt these are the cause of my unfortunate problem. :confused:

I don't know if one of my programs is interfering with the client or if their is an error in the programming.  Judging by the error, I'm guessing it is an error in the code.  If only I knew JavaScript. :sleepy:

**Comment by MarshallOfSound** [comment]
Does this error come with a stack trace?

Which version of the app are you running or are you running master?

**Comment by colingzesh** [comment]
I'm running version 3.0.1.  I don't know what a stack trace is.

**Comment by colingzesh** [comment]
I have a screenshot. [http://imgur.com/KK7VH6i](url)

**Comment by MarshallOfSound** [comment]
Hmm, without a stack trace (and you are right their isn't one there) the best I can suggest is for you to try installing the development environment which should give you a better error when you try to start it

**Comment by colingzesh** [comment]
Please excuse my ignorance.  I don't really know where I can install the development environment.

**Comment by MarshallOfSound** [comment]
Follow the development steps either in the README or in the CONTRIBUTING files

**Comment by FISHunderscore** [comment]
Note: Doing a system restore to a time before the error doesn't fix it, nor does reinstalling after the system restore.

**Comment by MarshallOfSound** [comment]
@FISHunderscore Can you try setting up the dev environment.  I've tried but can't replicate this issue

**Comment by colingzesh** [comment]
Okay.  I've been busy but I'll do it today.

**Comment by colingzesh** [comment]
I can feel you cringe at me asking this but... I don't understand the
Development instructions

**Comment by jostrander** [comment]
Can you uninstall or is that also causing the error? I'm going to attempt to reproduce this in a windows 8 VM since it's happened on both Win 10 and Win 7 I'm going to assume it isn't version specific.

**Comment by colingzesh** [comment]
I am able to uninstall it.  Reinstalling it gives me the same error message.

**Comment by jostrander** [comment]
Can you uninstall and then remove the `GPMDP`/ `Google Play Music Desktop Player` folder under  %appdata% as well as check and make sure there are no left over files in program files?

**Comment by jostrander** [comment]
There's also a squirrel setup log inside of `%localappdata%/GPMDP_3` that might have some clues as to what is happening, if it's in the set up area. If not, I would check this folder after uninstalling and make sure there are no residual files as well.

**Comment by FISHunderscore** [comment]
@jostrander I deleted %AppData%/Roaming/Google Play Music Desktop Player and %AppData%/Roaming/GPMDP, and reinstalling gives me this error.

![image](https://cloud.githubusercontent.com/assets/11418203/14089912/3d239578-f4ec-11e5-8deb-e329d704f9c8.png)

Log is [here](https://gist.github.com/FISHunderscore/4c23aaae8c4c193611dd).

**Comment by jostrander** [comment]
Getting a 404 on that log link.

**Comment by colingzesh** [comment]
I checked the log
2016-03-28 16:54:53> Program: Starting Squirrel Updater:
--updateSelf=C:\Users\Colin\AppData\Local\SquirrelTemp\Update.exe
2016-03-28 16:54:53> Program: About to wait for parent PID 4704
2016-03-28 16:59:21> Program: Starting Squirrel Updater:
2016-03-28 16:59:22> Program: Starting Squirrel Updater:
2016-03-28 16:59:23> Program: Starting Squirrel Updater:
2016-03-28 16:59:23> Program: Starting Squirrel Updater:

**Comment by colingzesh** [comment]
I wonder if uninstalling, deleting Google_Play_Music and GPMDP_3, and reinstalling will yield any new result

**Comment by jostrander** [comment]
@FISHunderscore it looks like the installer is having a hard time removing an existing file:

```
Failed to remove existing directory on full install, is the app still running???: System.UnauthorizedAccessException: Access to the path 'C:\Users\FISHunderscore\AppData\Local\GPMDP_3\app-3.0.1\d3dcompiler_47.dll' is denied.``

Specifically `C:\Users\FISHunderscore\AppData\Local\GPMDP_3\app-3.0.1\d3dcompiler_47.dll`

Can you try removing the `C:\Users\FISHunderscore\AppData\Local\GPMDP_3` directory and make sure no instance of gpmdp is running via task manager?

**Comment by FISHunderscore** [comment]
@jostrander I can kill the process (w/ task manager and Daphne), but it keeps coming back. Not sure why. I can't delete the directory while the task is running. I'm going to try rebooting (to see if it's in startup), and if needed booting in safe mode to delete the directory.

**Comment by jostrander** [comment]
@colingzesh It appears deleting GPMDP_3 fixes the issue, can you confirm it in your case?

**Comment by rfcsp** [comment]
I believe this is related. Haven't tried deleting and re-installing though.

![image](https://cloud.githubusercontent.com/assets/18138083/14102600/7bbc61dc-f593-11e5-9547-13b50460f66e.png)

**Comment by MarshallOfSound** [comment]
@rfcsp I would be interested in seeing that full stack trace.  Unfortunately the file name appears to be cut off :cry:

**Comment by jjerome00** [comment]
For what it's worth, I had the same error (Win10 x64).  I noticed a couple things which may have already been mentioned:
- The app fails to open and displays a small window (sorry I forget the contents)
- The process stays open, you have to open task manager and kill the process
- I deleted everything in the "Google Play Music Desktop Player" %appdata% folder.  Mine was at: C:\Users\<your username>\AppData\Roaming\Google Play Music Desktop Player\ 
- After this I started the app again and it worked!  There was another error message displayed (more on that below), but I closed the app and re-opened and everything seems fine.

I suspected it was the appdata folder since the app would not open after reboot.  I noticed in the comments that people seemed to be trying things out of order, so I thought I'd add what worked for me.  
...

The error on start after clearing my appdata folder is pasted below.  It may be unrelated but I figured I should be thorough:

An uncaught exception was reported.  
#### Info:
**OS:** win32
**Arch:** ia32
**GPMDP Version:** 3.0.1
**Time:** Tue Mar 29 2016 13:38:27 GMT-0700 (Pacific Daylight Time)

#### Error:
*Attempting to call a function in a renderer window that has been closed or released. Function provid*
```js
Error: Attempting to call a function in a renderer window that has been closed or released. Function provided here: Object.<anonymous> (C:\Users\jmj\AppData\Local\GPMDP_3\app-3.0.1\resources\app\build\inject\GPMInject\interface\mouseButtonNavigation.js:8:8.
    at BrowserWindow.callIntoRenderer (C:\Users\jmj\AppData\Local\GPMDP_3\app-3.0.1\resources\atom.asar\browser\lib\rpc-server.js:201:19)
    at emitTwo (events.js:92:20)
    at BrowserWindow.emit (events.js:172:7)
```
This issue was created automatically inside the `uncaughtException` handler

**Comment by FISHunderscore** [comment]
@jjerome00 Interesting, for you deleting GPMDP's folder in appdata/Roaming worked. For me, I had to delete from appdata/Roaming as well as appdata/Local.

Also, that error that you're getting seems to be super common right now. I believe it's been fixed already, with the fix coming out whenever 3.1.0 gets released.

**Comment by MarshallOfSound** [comment]
I believe from some code diving (no definite answer) that this issue was fixed in a bump of the Squirrel version used in the app
